tgui/packages/tgui/interfaces/ZoldorfPlayerShop/index.tsx

```diff
@@ -0,0 +1,128 @@
+/**
+ * @file
+ * @copyright 2024
+ * @author Glowbold (https://github.com/pgmzeta)
+ * @license MIT
+ */
+
+import { useBackend } from "../../backend";
+import { Box, Button, Flex, Section, Stack, Table } from "../../components";
+import { Window } from "../../layouts";
+import { ZoldorfPlayerShopData, ZoldorfProductListProps } from "./type";
+
+export const ZoldorfPlayerShop = (_, context) => {
+  const { act, data } = useBackend<ZoldorfPlayerShopData>(context);
+  const { soul_products, credit_products, credits } = data;
+  return (
+    <Window
+      width="500"
+      height="600"
+      fontFamily="Consolas"
+      font-size="10pt">
+      <Window.Content>
+        <Stack vertical fill minHeight="1%" maxHeight="100%">
+          <Stack.Item grow minHeight="1%" maxHeight="100%">
+            <Section fill scrollable height="100%">
```

Comment:
Going to have to explain all of the `1%`/`100%` things here. If these are just pulled from some other place (presumably vending machine code), please let me know where and I'll have to work out how to rewrite in a sane fashion.